### PR TITLE
Publishing to Maven sometimes needs a username and password.

### PR DESCRIPTION
**Pull request by shs96c**
It sometimes needs far more, but this is what we need to get
the Selenium project's jars published to the staging server.

### CODE REVIEW

File: src/com/facebook/buck/cli/PublishCommand.java [hunk 1]
```diff
@@ -75,6 +75,19 @@
...
+  @Option(
+      name = "--username",
+      aliases = "-u",
+      usage = "User name to use to authenticate with the server")
+  private String username;
+
+  @Option(
+      name = "--password",
+      aliases = "-p",
+      usage = "Password to use to authenticate with the server")
```
Comment (sdwilsh) [inline src/com/facebook/buck/cli/PublishCommand.java]: Some people don't like to pass in passwords on the command line (it shows up in history).  Is there a way to provide an option to prompt for it too (I'm not sure if buckd is going to rain heavily on this idea).

Comment (LegNeato) [inline src/com/facebook/buck/cli/PublishCommand.java]: Perhaps we copy what the AWS stuff does? It has a multi-layered approach, starting with a file in the homedir, moving to ENV variables, and then finally using values specified on the command line. We could even do he whole `AWS_PROFILE` jazz for the homedir stuff.

Comment (shs96c) [inline src/com/facebook/buck/cli/PublishCommand.java]: We could, but I'd prefer to do that in a follow-up diff.

**Comment by sdwilsh** [comment]
To be clear, my comment was a drive-by.  I'm too far removed from Buck's day-to-day development to consider myself qualified to review it, so someone else from the team should look at it.

**Pull request closed without merging**